module_analysis/2_add_annotations.py

```diff
@@ -0,0 +1,246 @@
+pa14_gene_annot_filename = paths.GENE_PA14_ANNOT
+
+pao1_gene_annot = pd.read_csv(pao1_gene_annot_filename, index_col=0, header=0)
+pa14_gene_annot = pd.read_csv(pa14_gene_annot_filename, index_col=0, header=0)
+# -
+
+pao1_gene_annot = pao1_gene_annot["Name"].to_frame("gene name")
+pa14_gene_annot = pa14_gene_annot["Name"].to_frame("gene name")
+
+# ## Add module labels
+
+# Add module labels
+pao1_gene_module_labels = pao1_membership.merge(
+    pao1_module_labels, left_on="module id", right_index=True
+)
+pa14_gene_module_labels = pa14_membership.merge(
+    pa14_module_labels, left_on="module id", right_index=True
+)
+
+# ## Add gene names
+
+# Add gene names
+pao1_gene_module_labels = pao1_gene_module_labels.merge(
+    pao1_gene_annot, left_index=True, right_index=True
+)
+pa14_gene_module_labels = pa14_gene_module_labels.merge(
+    pa14_gene_annot, left_index=True, right_index=True
+)
+
+print(pao1_gene_module_labels.shape)
+pao1_gene_module_labels.head()
+
+print(pa14_gene_module_labels.shape)
+pa14_gene_module_labels.head()
+
+# ## Add KEGG pathways
+#
+# For each pathway, what genes are contained in it
+
+# +
+pao1_pathway_filename = "https://raw.githubusercontent.com/greenelab/adage/master/Node_interpretation/pseudomonas_KEGG_terms.txt"
+
+pao1_pathways = pd.read_csv(pao1_pathway_filename, sep="\t", index_col=0, header=None)
+# -
+
+pao1_pathways[2] = pao1_pathways[2].str.split(";").apply(set)
+pao1_pathways.index = pao1_pathways.index.str.split(" - ").str[0]
+pao1_pathways.head()
+
+gene_to_pathways_df = pd.DataFrame(
+    index=pao1_gene_module_labels.index, columns=list(pao1_pathways.index)
+)
+
+# %%time
+for gene in gene_to_pathways_df.index:
+    gene_to_pathways_df.loc[gene] = [
+        gene in pao1_pathways.loc[pathway, 2] for pathway in pao1_pathways.index
+    ]
+
+# ## Add operon
+#
+# For each operon, what genes are contained in it
+#
+# NOTE: This code takes a while to run so for now its commented out
+
+# +
+pao1_operon_filename = "https://raw.githubusercontent.com/greenelab/core-accessory-interactome/6635c0e357c0172c2cebd0368648030e0ee4beaf/data/metadata/operons_format.csv"
+
+pao1_operons = pd.read_csv(pao1_operon_filename, index_col=0, header=0)
+# -
+
+pao1_operons.head()
+
+pao1_operons["Genes"] = pao1_operons["Genes"].str.split(";").apply(set)
+pao1_operons.head()
+
+# Remove operons with a single gene
+pao1_operons = pao1_operons[pao1_operons["Genes"].apply(len) > 1]
+
+gene_to_operons_df = pd.DataFrame(
+    index=pao1_gene_module_labels.index, columns=list(pao1_operons.index)
+)
+
+# %%time
+for gene in gene_to_operons_df.index:
+    gene_to_operons_df.loc[gene] = [
+        gene in pao1_operons.loc[operon, "Genes"] for operon in pao1_operons.index
+    ]
+
+# Add operons to pathway annotations
+pao1_gene_annot = gene_to_pathways_df.merge(
+    gene_to_operons_df, left_index=True, right_index=True, how="outer"
+)
+
+print(pao1_gene_annot.shape)
+pao1_gene_annot.head()
```

Comment:
```suggestion
pao1_gene_annot.head().T
```
I recommend transposing the dataframe head so we can see the field names better.

Reply:
I see what you're saying. For now, I think I'll keep the columns as annotations because it'll be easier for our collaborators to sort when they are searching through these results